src/databricks/labs/ucx/workspace_access/manager.py

```diff
@@ -144,6 +148,43 @@ def apply_group_permissions(self, migration_state: MigrationState) -> bool:
         logger.info("Permissions were applied")
         return True
 
+    def apply_group_permissions_private_preview_api(self, migration_state: MigrationState) -> bool:
+        if len(migration_state) == 0:
+            logger.info("No valid groups selected, nothing to do.")
+            return True
+        logger.info(
+            f"Applying the permissions to account groups. "
+            f"Total groups to apply permissions: {len(migration_state)}."
+        )
+        errors: list[Exception] = []
+        for migrated_group in migration_state.groups:
+            errors.extend(self._migrate_group_permissions_paginated(migrated_group))
+        if len(errors) > 0:
+            logger.error(f"Detected {len(errors)} errors while applying permissions")
+            raise ManyError(errors)
+        logger.info("Permissions were applied")
+        return True
+
+    def _migrate_group_permissions_paginated(self, migrated_group: MigratedGroup) -> list[Exception]:
+        batch_size = 1000
+        errors: list[Exception] = []
+        while True:
+            try:
+                response = self._ws.permission_migration.migrate_permissions(
+                    self._ws.get_workspace_id(),
+                    migrated_group.name_in_workspace,
+                    migrated_group.name_in_account,
+                    size=batch_size,
+                )
+                if response.permissions_migrated is None:
+                    break
+                if response.permissions_migrated < batch_size:
```

Comment:
```suggestion
                if response.permissions_migrated == 0:
```
perhaps this is better? also do some logging.

Reply:
done